### DISCUSSION TITLE
batch que please?

### DISCUSSION THREAD

**Discussion by unifirer** [erew123/alltalk_tts#11] [General]
with the deepspeed optimisation, we can actually do multiple books in 1 night!

if u have some spare time, a batch que would allow us to do so!

another fantastic reason to do so, somewhere in the process, the max length of wav is <17h. after that the output shows 17 but is 4.5hrs max, maybe its a limit of ffmpeg or something else like a limit of the wav file or the header or something, i couldnt find it tho

thank you very much

**Comment by erew123** [comment]
Well there is a 4GB limitation https://en.wikipedia.org/wiki/WAV#Limitations and it is an old standard, going back to 1991. I cant speak as to how things such as ffmpeg would handle such large files. But that's 6.8 hours at 44100Hz, so maybe closer to 10-12 tops at 22050Hz (with headers/padding overheads). 

An average book, you're looking at generating 70,000-100,000 words, 3000-6000 sentences (or wav files should I say) then trying to combine them into 1x file. Its quite a technical overhead on a system to do that and I'm not sure where the limitations are. It would be a decent amount of testing and coding to make something like that work.

Can I ask what your exact use case is?

**Comment by unifirer** [comment]
Oh that makes sense. it outputs 24k and does 9.5 hours without a problem. So the maximum is somewhere between 10 and 17 hours.

The final combination process was so intensive that my game lagged out and my computer froze  for 2minutes.

Audiobook production. That's my use case.  If you have spare time and energy to make it output multiple files instead in a queue as a batch as a new feature request. That would help a lot.

Like the feature that handbrake has. batch processing.

Please inform me of interesting jobs and opportunities
________________________________

**Comment by unifirer** [comment]
Perhaps instead of queing multiple videos. que multiple text files


________________________________

**Comment by unifirer** [comment]
With text files we can split them into chapters too!
________________________________

**Comment by erew123** [comment]
Ok. Well its not a core feature that has been in my roadmap and it would take some thinking around how best to split out that much text, send bits off for generation, potentially combine every X amount of generated wavs, then keep going and do a smaller combine at the end. Compiling that much audio into one and how to handle that process is a potentially complicated task. And obviously, the only way I could test the robustness of such a thing would be to write code then set a machine off and running for X hours and if it crashes/errors, change the code and then send it off again etc. 

I also don't want to get into feature creep while working on what is a new project, as I do have certain goals.. Ultimately, it would probably be best handled by a separate script that calls on the AllTalk API. It can deal with any filtering/cleaning, breaking down of text ahead of time and you could probably feed in a text file or something. 

So, Im not saying no, but on the same note, I've got quite a bit going on currently to jump off to batch queue text generation and I need to get those core things done initially.

Ill move this over from issues to discussions so I've got a reference there. Ill mull the problem over in my head and see where I get to with my other bits of the roadmap.

Also that means if others discover this conversation, they are welcome to chip in, if its an idea they are interested in.

**Comment by unifirer** [comment]
Thank you for your hard work

**Comment by erew123** [comment]
![image](https://github.com/erew123/alltalk_tts/assets/35898566/17759973-12c7-4156-8e68-7fad7cc5b293)

Almost done.... Not available yet!

If you're going to be making money from this... I hope you'll make me a donation at some point ;)

**Comment by unifirer** [comment]
Thank you very much. I haven't made a cent yet because in every hour of speech generated. I had around 5 seconds of continuous screech or rumbles.  

  
  I've managed to get it down to 1 second after decreasing the parameters to the lower values. Thank you for that function too
  

  
Once I start making a profit you'll get a percentage share for sure. I'm still not sure about the demand yet. But the market seems to be way overpriced right now.   
  
 On 7/01/2024 at 5:58 am, erew123  ***@***.***>  wrote:
  
 image.png (view on web)
 Almost done.... Not available yet!
 If you're going to be making money from this... I hope you'll make me a donation at some point ;)

**Comment by unifirer** [comment]
> ![image](https://private-user-images.githubusercontent.com/35898566/294703554-17759973-12c7-4156-8e68-7fad7cc5b293.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1Nzc0NjQsIm5iZiI6MTcwNDU3NzE2NCwicGF0aCI6Ii8zNTg5ODU2Ni8yOTQ3MDM1NTQtMTc3NTk5NzMtMTJjNy00MTU2LThlNjgtN2ZhZDdjYzViMjkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA2VDIxMzkyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZjIwNmI0Yjg2ZjZmMmZhZjcwZmQxNjI0OWVkZTI1ZGEzNjFkY2YyZDgxNjUwMzc5Y2U4NzFiYTdmOGVmYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.46P0RsnfbuhRqoss4pBnwaP8aeT0cBcM1tOpPHNhM5c)
> 
> Almost done.... Not available yet!
> 
> If you're going to be making money from this... I hope you'll make me a donation at some point ;)

i tried to make a donation when i first started using your app and tried again now, cant see a link, do u have it somewhere else? its not on ur profile

**Comment by erew123** [comment]
Thanks! Though, you're right, I've got no links at the moment for anything like that... I've yet to figure it out! Guess Ive been too busy coding!

Here you go though! https://github.com/erew123/alltalk_tts#-alltalk-tts-generator

I'm going to mark this closed now hah!
